pkg/util/ec2/ec2.go

```diff
@@ -76,6 +76,18 @@ func GetLocalIPv4() ([]string, error) {
 	return []string{ip}, nil
 }
 
+// GetPublicIPv4 gets the public IPv4 for the currently running host using the EC2 metadata API.
+func GetPublicIPv4() (string, error) {
+	if !config.IsCloudProviderEnabled(CloudProviderName) {
+		return "", fmt.Errorf("cloud provider is disabled by configuration")
+	}
+	ip, err := getMetadataItem("/public-ipv4")
+	if err != nil {
+		return "", err
+	}
+	return ip, nil
```

Comment:
No need to force set ip to `""`. we can directly `return getMetadataItem("/public-ipv4")`